core/serv/src/main.rs

```diff
@@ -136,35 +141,56 @@ impl TryFrom<&CliArgs> for CliCtx {
     }
 }
 
+#[derive(Clone)]
 struct ServiceContext {
-    db: DbExecutor,
-    data_dir: PathBuf,
+    dbs: HashMap<TypeId, DbExecutor>,
+    default_db: DbExecutor,
 }
 
-impl<Service> Provider<Service, DbExecutor> for ServiceContext {
+impl<Service: 'static> Provider<Service, DbExecutor> for ServiceContext {
     fn component(&self) -> DbExecutor {
-        self.db.clone()
+        let service_type = TypeId::of::<Service>();
+        if let Some(db) = self.dbs.get(&service_type) {
+            db.clone()
+        } else {
+            self.default_db.clone()
+        }
     }
 }
 
-impl<Service> Provider<Service, PathBuf> for ServiceContext {
-    fn component(&self) -> PathBuf {
-        self.data_dir.clone()
+impl ServiceContext {
+    fn make_entry<S: 'static>(path: &PathBuf, name: &str) -> Result<(TypeId, DbExecutor)> {
+        Ok((TypeId::of::<S>(), DbExecutor::from_data_dir(path, name)?))
+    }
+
+    fn from_data_dir(path: &PathBuf, name: &str) -> Result<Self> {
+        let default_db = DbExecutor::from_data_dir(path, name)?;
+        default_db.apply_migration(ya_persistence::migrations::run_with_output)?;
```

Comment:
This line will add activity migrations to all databases. In my code there were no migrations done here excluding default database. Each service runs migrations by itself.

Reply:
Maybe a little more explanation: activity database uses migrations defined in ya-persistence. But all other services have their own migrations sets.

Reply:
I'll move `ya-persistence::migrations` to `ya-activity`

Reply:
It's probably good choice